BitList.cs

```diff
@@ -0,0 +1,1737 @@
+                {
+                    intList.Add(num);
+                }
+            }
+
+            this.bits = new BitArray(intList.ToArray());
+        }
+
+        /// <summary>
+        /// Ctor.
+        /// </summary>
+        /// <param name="bits">A 16 bit character.</param>
+        public BitList(char bits)
+        {
+            this.bits = new BitArray(BitConverter.GetBytes(bits));
+        }
+
+        /// <summary>
+        /// Ctor.
+        /// </summary>
+        /// <param name="bits">A 16 bit character array (<see cref="char"/> array).</param>
+        public BitList(char[] bits)
+        {
+            //Convert the Char array to a Byte array and then create a new BitArray instance.
+            var bytesList = new List<byte[]>();
+            foreach (var num in bits)
+            {
+                bytesList.Add(BitConverter.GetBytes(num));
+            }
+
+            var byteList = new List<byte>();
+            foreach (var bytes in bytesList)
+            {
+                foreach (var num in bytes)
+                {
+                    byteList.Add(num);
+                }
+            }
+
+            this.bits = new BitArray(byteList.ToArray());
+        }
+
+        /// <summary>
+        /// Ctor.
+        /// </summary>
+        /// <param name="bits">A 16 bit character array (<see cref="string"/>).</param>
+        public BitList(string bits)
+        {
+            this.bits = new BitList(bits.ToCharArray()).bits;
+        }
+
+        /// <summary>
+        /// Ctor.
+        /// </summary>
+        /// <param name="bits">A 16 bit character array of array.</param>
+        public BitList(string[] bits)
+        {
+            var bitList = new BitList();
+            foreach (var str in bits)
+            {
+                bitList += new BitList(str);
+            }
+
+            this.bits = bitList.bits;
+        }
+
+        bool ICollection<bool>.IsReadOnly => false;
+
+        /// <summary>
+        /// This property return true if the instance has no values.
+        /// </summary>
+        public bool IsEmpty => bits == null || bits.Length == 0;
+
+        /// <summary>
+        /// The amount of bits of the instance.
+        /// </summary>
+        public int Count => IsEmpty ? 0 : bits.Length;
+
+        /// <summary>
+        /// Indexer.
+        /// </summary>
+        public bool this[int index]
+        {
+            get => Get(index);
+            set => Set(index, value);
+        }
+
+        /// <summary>
+        /// Indexer.
+        /// </summary>
+        /// <param name="index">Start index.</param>
+        /// <param name="count">Count of bits.</param>
+        public BitList this[int index, int count]
+        {
+            get => Get(index, count);
+            set
+            {
+                if (count >= value.Count)
+                {
+                    Set(index, value);
+                }
+                else if (count < value.Count)
```

Comment:
Codacy found an issue: [Add the missing 'else' clause.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=5559704)